### DISCUSSION TITLE
IronPython 3.0

### DISCUSSION THREAD

**Discussion by kookma** [DanWBR/dwsim#153] [Ideas]
As the most common Python version nowadays used is Python 3, I may recommend to get ready for https://github.com/IronLanguages/ironpython3 and replace the current Ironpython 2 in DWSIM distribution in near future!

One question: I believe Ironpython is used for simple scripting in DWSIM and it is not to use for CAPE-OPEN and write complex codes in Python like a fluidized bed reactions! If so, it is good to clear this in documentation!

The Python.Net is used for bridging to existing Python installation and one can use the full power of python eg link to SciPy, Tensorflow, ... So, we need to have good support in this part for most recent release of Python!

**Comment by DanWBR** [comment]
@kookma think about IronPython as a general-purpose scripting language not tied to a specific Python version, but based on one. It doesn't matter if it is IronPython 2 or 3 (which is far from being usable, as the developers say). What is important is that the current version is more than enough for what it is up to. IronPython will not be updated because changing its version from 2 to 3 would break all existing scripts, for no particular reason.

IronPython is not supposed to replicate Python behavior in the context of DWSIM scripting capabilities, but rather to provide a scripting environment that is easy to use and based on a language that most people will find easy and productive to work with.

Python.NET is there to integrate with external python environments, and it evolves together with Python. One must always think about backward compatibility with existing scripts from other users. In my opinion, this is more important than being always up-to-date.

I do not agree with your sentence about IronPython not being able to handle complex code. In my opinion, it is exactly the opposite. IronPython is much more powerful for all DWSIM purposes because it supports all .NET features and its vast library of components, and runs much faster than Python.NET, and is much easier to debug.

**Reply by kookma** [reply]
Hi @DanWBR 
Thank you for clarification! So we stick to Ironpython 2.7 for future as it is the most stable and is the scripting environment for DWSIM! That makes sense!

For complex third party development like development AI models which needs third party python libraries (like Tensorflow, ...) or where we need Scipy and Numba, ... DWSIM will bridge to Python 3.9!
